Update title only when the editor wizard is closed

fjorgemota opened the pull request:
Fixes #5240

### Changes proposed in this Pull Request

* Update title only when the editor wizard is closed, instead of updating it on real time;
* Change reference from `data` to `wizardData`; 
* Use `title` instead of `courseTitle`/`lessonTitle`, and `description` instead of `courseDescription`;

### Testing instructions

1. Create a new course, edit the title and the description and guarantee that nothing updates in the background;
2. a) Close the editor wizard, and verify that the title, description and excerpt appear correctly in the correct places;
2. b) Select a pattern and again verify that the title, description and excerpt appear correctly in the correct places;

In the same sense:

1. Create a new lesson, edit the title and guarantee that nothing updates in the background;
2. a) Close the editor wizard, and verify that the title appear correctly in the correct place;
2. b) Select a pattern and again verify that the title appear correctly in the correct place;

fjorgemota committed:
Update title only when the editor wizard is closed

So we have some consistency between title and description

fjorgemota committed:
Fix tests of course and lesson details step

fjorgemota committed:
Don't update post excerpt if we don't have a defined courseDescription

aaronfc reviewed (dismissed):
Works well!

I don't like it very much that the "editor-wizard-modal" knows about the course or lesson attributes... but I can't think of any elegant solution for that.

renatho commented:
> I don't like it very much that the "editor-wizard-modal" knows about the course or lesson attributes

I see it. Maybe we could use it in a more generic way? We could just have the names as `title` and `description` regardless the post type.

I also take the opportunity to bring back the discussion about the `data / wizardData`, and `setData: setWizardData`. I think we could set the prop names as `wizardData` and `setWizardData` directly instead changing the name in the components only. But not a requisite for this PR if we consider it big, since it's not related.

fjorgemota commented:
I feel the same, @aaronfc and @renatho, and I think we can really change the name of the attributes to sound more generic.

About renaming the props, that doesn't seem a big task at all IMHO. I can change that too without any issues.

I'm just not sure about what to do considering the discussion related to this PR vs. the code freeze. More details here: p1655127739832199/1654886363.140779-slack-C8QJAM52B

WDYT?

fjorgemota committed:
Rename fields in the editor wizard to be more generic

So "lessonTitle"/"courseTitle" becomes "title" and "courseDescription" becomes "description".

fjorgemota committed:
Rename data to wizardData

To avoid renaming the instances all the time

renatho commented on assets/admin/editor-wizard/editor-wizard-modal.js:
```diff
@@ -18,20 +18,25 @@ import '../../shared/data/api-fetch-preloaded-once';
 	const steps = useEditorWizardSteps();
 	const setDefaultPattern = useSetDefaultPattern( {
-		'sensei-content-description': dataState[ 0 ].courseDescription,
+		'sensei-content-description': wizardDataState[ 0 ].description,
```
Since we're accessing the `wizardDataState[ 0 ]` many times now, we could make it clearer by creating a new variable like `const wizardData = wizardDataState[ 0 ];`. WDYT?

renatho reviewed (dismissed):
Thank you for the changes!
Looks good and works well!

I just added a small suggestion. Feel free to skip it, or merge it directly after updating it, since it's a very small change. ;)

fjorgemota committed:
Remove unneeded wizardData from CurrentStep.Actions

fjorgemota committed:
Use wizardData to refer to wizardDataState[0] in editor wizard modal

alexsanford reviewed (changes requested):
Let's hold off on this one for now. Looks like we want to keep the real-time title update at the moment. Maybe when @Luchadores gets back we can discuss whether this is important from a UX perspective.

cc @burtrw

fjorgemota commented:
No problem, @alexsanford. I am sending a new PR to add only the refactor based on the suggestions provided by @renatho, then: https://github.com/Automattic/sensei/pull/5269

fjorgemota commented:
Closing considering that #5240 was closed as not planned.
